lib/mrtrix3/run.py

```diff
@@ -335,3 +338,52 @@ def versionMatch(item):
 
   app.error('Unable to find executable for MRtrix3 command ' + item)
 
+
+
+# If the target executable is not a binary, but is actually a script, use the
+#   shebang at the start of the file to alter the subprocess call
+def _shebang(item):
+  import os
+  from mrtrix3 import app
+  from distutils.spawn import find_executable
+  # If a complete path has been provided rather than just a file name, don't perform any additional file search
+  if os.sep in item:
+    path = item
+  else:
+    path = versionMatch(item)
+    if path == item:
+      path = find_executable(exeName(item))
+  if not path:
+    app.debug('File \"' + item + '\": Could not find file to query')
+    return []
+  # Read the first 1024 bytes of the file
+  with open(path, 'rb') as f:
+    data = f.read(1024)
+  # List of permissible text characters
+  textchars = bytearray({7,8,9,10,12,13,27} | set(range(0x20, 0x100)) - {0x7f})
+  # Are there any non-text characters? If so, it's a binary file, so don't go looking for a shebang
```

Comment:
I probably wouldn't bother checking so hard. There is an outside chance that the script might have been written with a comment in some non-English language in UTF-8 or something, which will cause this fail. I would just stick with detecting a `#!`: no binary executable will start with that. On Linux, we'd have ELF format binaries, which will start with (0x45)ELF, it'll be something equally obscure on Windows. I think it would be sufficient to assume that a file starting with `#!` is a script, without checking whether the rest is interpretable as text.